edb/pgsql/common.py

```diff
@@ -186,6 +186,18 @@ def convert_name(name, suffix='', catenate=True):
         return schema, dbname
 
 
+def update_aspect(name, aspect):
+    """Update the aspect on a non catenated name.
+
+    It also needs to be from an object that uses ids for names"""
+    suffix = get_aspect_suffix(aspect)
+    stripped = name[1].rsplit("_", 1)[0]
```

Comment:
Ugh, it feels like we shouldn't be doing this, but the whole `PointerStorageInfo` thing was in a dire need of a refactor for a while, so it's a fair hack.

Reply:
I think the particular approach of slicing up the string is pretty unfortunate, but I like having the PointerStorageInfo drive getting table names for this sort of thing. (Otherwise I would have needed to duplicate some logic that was already done when computing the PointerStorageInfo table_name). I think there are a bunch of other places that could get the info from there too.

Reply:
I didn't mean that we should do away with PointerStorageInfo, just modernize it a bit.  Like, make `table_name()` a callable accepting aspect or add another method.